src/main/java/com/nextcloud/android/sso/Constants.java

```diff
@@ -22,7 +22,7 @@
 public class Constants {
 
     // Authenticator related constants
-    public static final String SSO_USERNAME = "username";
+    public static final String SSO_USER_ID = "user_id";
```

Comment:
Changing this line breaks backward compatibility, right? Is there another option to make the transition soft? And we would have to upgrade third party apps first before you release this into the nextcloud files app.

Maybe add something like this: (in the `AccountImporter.java`)

```java
String userId = future.getString(Constants.SSO_USER_ID);
if (userId == null) {
  // backwards compatibility
  userId = future.getString("username");
}
```

Reply:
This is a nice idea.